verify-matching-service-test-tool/src/main/java/uk/gov/ida/verifymatchingservicetesttool/configurations/ConfigurationReader.java

```diff
@@ -9,22 +9,26 @@
 
 public class ConfigurationReader {
 
-    public static ApplicationConfiguration getConfiguration() {
+    private static final String VERIFY_MATCHING_SERVICE_TEST_TOOL_YML = "verify-matching-service-test-tool.yml";
+
+    public static ApplicationConfiguration getConfiguration(String configFileLocation) {
         try {
-            return new ObjectMapper(new YAMLFactory()).readValue(
-                new File(getConfigurationFolderLocation()),
-                ApplicationConfiguration.class
-            );
+            return new ObjectMapper(new YAMLFactory()).readValue(new File(getConfigurationFolderLocation(configFileLocation)),
+                    ApplicationConfiguration.class);
         } catch (IOException e) {
             throw new RuntimeException(e);
         }
     }
 
-    private static String getConfigurationFolderLocation() {
+    private static String getConfigurationFolderLocation(String configFileLocation) {
+        return configFileLocation != null ? configFileLocation : getDefaultConfigLocation();
```

Comment:
I think `Optional.ofNullable(configFileLocation).orElseGet(() => getDefaultConfigLocation())` is more idiomatic Java 8

Reply:
Done